library/pod-security-policy/flexvolume-drivers/template.yaml

```
@@ -36,37 +36,59 @@ spec:
                     type: string
   targets:
     - target: admission.k8s.gatekeeper.sh
-      rego: |
-        package k8spspflexvolumes
+      code:
+      - engine: K8sNativeValidation
+        source:
+          variables:
+          - name: flexVolumes
+            expression: |
+              !has(variables.anyObject.spec.volumes) ? [] :
+                variables.anyObject.spec.volumes.filter(volume, has(volume.flexVolume))
+          - name: allowedFlexVolumeDrivers
+            expression: variables.params.allowedFlexVolumes.map(volume, volume.driver)
+          - name: badFlexVolumeNames
+            expression: |
+              variables.flexVolumes.map(volume,
```

```suggestion
              !has(variables.flexVolumes) ? [] :  variables.flexVolumes.map(volume,
```

to avoid null pointers